packages/block-library/src/navigation/index.php

```diff
@@ -67,6 +67,513 @@ function block_core_navigation_sort_menu_items_by_parent_id( $menu_items ) {
 	}
 }
 
+/**
+ * Helper functions used to render the navigation block.
+ */
+class WP_Navigation_Block {
+	/**
+	 * Returns whether or not this is responsive navigation.
+	 */
+	private static function is_responsive_navigation( $attributes ) {
+		/**
+		 * This is for backwards compatibility after `isResponsive` attribute has been removed.
+		 */
+
+		$has_old_responsive_attribute = ! empty( $attributes['isResponsive'] ) && $attributes['isResponsive'];
+		return isset( $attributes['overlayMenu'] ) && 'never' !== $attributes['overlayMenu'] || $has_old_responsive_attribute;
+	}
+
+	/**
+	 * Returns whether or not a navigation has a submenu.
+	 */
+	private static function does_navigation_have_submenus( $inner_blocks ) {
```

Comment:
My main concern is that this function is now called several times so its probably slightly less performant.

Reply:
We could exit the loop early and return from the method if a submenu is found. I think `continue` will do that 🤔

Reply:
> now called several times

Why is this different than before?

> probably slightly less performant.

Also this is likely to be marginal. Is it a big concern?

Reply:
Let's create an instance variable to cache the result of the function. Then we can check if the instance variable has already been assigned and exit early with that result.

Reply:
> Why is this different than before?
Before it was called in the render function and saved in a variable.